Source/FakeItEasy.IntegrationTests/DummyTests.cs

```diff
@@ -38,6 +39,33 @@ public void Correct_exception_should_be_thrown_when_dummy_is_requested_for_non_f
             }
         }
 
+        [SuppressMessage("Microsoft.Naming", "CA2204:Literals should be spelled correctly", MessageId = "CanCreateDummyOfType",
+            Justification = "That's really the name of the method.")]
+        [Test]
+        public void Definition_for_type_should_be_cached_after_first_use()
+        {
+            var firstDummy = A.Dummy<Dummy>();
+
+            var secondDummy = A.Dummy<Dummy>();
+
+            Assert.That(secondDummy.Count, Is.EqualTo(firstDummy.Count), "CanCreateDummyOfType was called the wrong number of times.");

```

Comment:
Regrettably, no. I'd've preferred it as well. 

`CanCreateDummyOfType` will be called on a dummy definition when we're looking for any (not yet cached) dummy definition to use. So if other tests have already tried to make dummies, there's a chance that the framework will have already called `CanCreateDummyOfType`.

Right now if I make the change, it fails, as the method has been called 5 times, not 1.

Of course, as I type this, I realize that I can reset the count in the test. Two secs.

```csharp
[Test]
public void CanCreateDummyOfType_should_not_be_called_again_after_definition_for_a_dummy_type_is_found()
{
    // Clear the count, since CanCreateDummyOfType may have been called for other types
    // in other tests.
    DummyTestsDummyDefinition.TimesCanCreateDummyOfTypeWasCalled = 0;

    A.Dummy<Dummy>();
    DummyTestsDummyDefinition.TimesCanCreateDummyOfTypeWasCalled.Should()
        .Be(1, "we need to call CanCreateDummyForType to create the Dummy");

    A.Dummy<Dummy>();

    DummyTestsDummyDefinition.TimesCanCreateDummyOfTypeWasCalled.Should()
        .Be(1, "the definition should've been cached, so there's no need to call CanCreateDummyForType");
}
```

The first  `Should` is practically a precondition test, which I'm not always so keen on. Say the word and I'll strike it.